src/nsolid/nsolid_api.cc

```diff
@@ -2360,34 +2360,94 @@ void BindingData::PushSpanDataUint64Impl(BindingData* data,
+                                         const FastOneByteString& val) {
+  PushSpanDataStringImpl(FromJSObject<BindingData>(receiver),
+                         trace_id,
+                         type,
+                         std::string(val.data, val.length));
+}
+
+
+void BindingData::PushSpanDataStringImpl(BindingData* data,
+                                         uint32_t trace_id,
+                                         uint32_t type,
+                                         const std::string& val) {
+  Span::PropType prop_type = static_cast<Span::PropType>(type);
+  auto prop = Span::createSpanProp<std::string>(prop_type, val);
+  EnvInst* envinst = data->env()->envinst_.get();
+  EnvList::Inst()->GetTracer()->pushSpanData(
+      SpanItem{ trace_id, envinst->thread_id(), std::move(prop) });
+}
+
+
+void BindingData::SlowPushSpanDataString3(
+    const FunctionCallbackInfo<Value>& args) {
+  Isolate* isolate = args.GetIsolate();
+  DCHECK_EQ(args.Length(), 5);
+  DCHECK(args[0]->IsUint32());
+  DCHECK(args[1]->IsUint32());
+  DCHECK(args[2]->IsString());
+  DCHECK(args[3]->IsString());
+  DCHECK(args[4]->IsString());
+  uint32_t trace_id = args[0].As<Uint32>()->Value();
+  uint32_t type = args[1].As<Uint32>()->Value();
+  Local<String> value_s1 = args[2].As<String>();
+  Local<String> value_s2 = args[3].As<String>();
+  Local<String> value_s3 = args[4].As<String>();
+  BindingData* data = FromJSObject<BindingData>(args.This());
+  const std::string val1 = *String::Utf8Value(isolate, value_s1);
+  const std::string val2 = *String::Utf8Value(isolate, value_s2);
+  const std::string val3 = *String::Utf8Value(isolate, value_s3);
+  PushSpanDataStringImpl3(data, trace_id, type, val1, val2, val3);
+}
+
+
+void BindingData::FastPushSpanDataString3(v8::Local<v8::Object> receiver,
+                                          uint32_t trace_id,
+                                          uint32_t type,
+                                          const FastOneByteString& val1,
+                                          const FastOneByteString& val2,
+                                          const FastOneByteString& val3) {
+  PushSpanDataStringImpl3(FromJSObject<BindingData>(receiver),
+                         trace_id,
+                         type,
+                         std::string(val1.data, val1.length),
+                         std::string(val2.data, val2.length),
+                         std::string(val3.data, val3.length));
+}
+
+void BindingData::PushSpanDataStringImpl3(BindingData* data,
+                                          uint32_t trace_id,
+                                          uint32_t type,
+                                          const std::string& val1,
+                                          const std::string& val2,
+                                          const std::string& val3) {
+  Span::PropType prop_type = static_cast<Span::PropType>(type);
+  ASSERT_EQ(prop_type, Span::kSpanHttpReqUrl);
```

Comment:
Shouldn't it be a `DCHECK` instead? I wonder if we should crash the app if malformed span data is sent. Technically, if that happens, it is our fault, not the user's fault, to have a crash in there.